[NNC] moved lowerings out of the TensorExprKernel and into independent functions

Chillee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56718 [WIP] Added skeleton of python dispatch key
* #56875 [nnc] ported conv2d lowering over
* #56775 [NNC] ported over sum and softmax to new scheme
* #56766 [nnc] ported some more ops + added vectors to argvalue
* **#56679 [NNC] moved lowerings out of the TensorExprKernel and into independent functions**



moved lowerings out of the TensorExprKernel and into independent functions

Differential Revision: [D28082921](https://our.internmc.facebook.com/intern/diff/D28082921)

Chillee committed:
moved lowerings out of the TensorExprKernel and into independent functions

[ghstack-poisoned]

Chillee committed:
Update on "moved lowerings out of the TensorExprKernel and into independent functions"




[ghstack-poisoned]

Chillee committed:
Update on "moved lowerings out of the TensorExprKernel and into independent functions"




[ghstack-poisoned]

Chillee committed:
Update on "moved lowerings out of the TensorExprKernel and into independent functions"




[ghstack-poisoned]

Chillee committed:
Update on "moved lowerings out of the TensorExprKernel and into independent functions"




[ghstack-poisoned]

Chillee committed:
Update on "moved lowerings out of the TensorExprKernel and into independent functions"




[ghstack-poisoned]

Chillee committed:
Update on "moved lowerings out of the TensorExprKernel and into independent functions"




moved lowerings out of the TensorExprKernel and into independent functions

[ghstack-poisoned]

Chillee committed:
Update on "moved lowerings out of the TensorExprKernel and into independent functions"




moved lowerings out of the TensorExprKernel and into independent functions

[ghstack-poisoned]

Chillee committed:
Update on "moved lowerings out of the TensorExprKernel and into independent functions"




moved lowerings out of the TensorExprKernel and into independent functions

[ghstack-poisoned]

Chillee committed:
Update on "moved lowerings out of the TensorExprKernel and into independent functions"




moved lowerings out of the TensorExprKernel and into independent functions

[ghstack-poisoned]

Chillee committed:
Update on "moved lowerings out of the TensorExprKernel and into independent functions"




moved lowerings out of the TensorExprKernel and into independent functions

[ghstack-poisoned]

Chillee committed:
Update on "moved lowerings out of the TensorExprKernel and into independent functions"




moved lowerings out of the TensorExprKernel and into independent functions

[ghstack-poisoned]

ZolotukhinM commented on torch/csrc/jit/tensorexpr/tensorexpr_init.cpp:
```diff
@@ -535,6 +535,41 @@ void initTensorExprBindings(PyObject* module) {
+        std::vector<ArgValue> argInputs;
+        for (auto inp : inputs) {
+          if (py::isinstance<Placeholder>(inp)) {
+            // std::cout << "placeholder" << std::endl;
```
Nit: leftover here and in similar places below.

ZolotukhinM reviewed (approved):
Looks good!

Chillee committed:
Update on "[NNC] moved lowerings out of the TensorExprKernel and into independent functions"




moved lowerings out of the TensorExprKernel and into independent functions

[ghstack-poisoned]

Chillee committed:
Update on "[NNC] moved lowerings out of the TensorExprKernel and into independent functions"




moved lowerings out of the TensorExprKernel and into independent functions

Differential Revision: [D28082921](https://our.internmc.facebook.com/intern/diff/D28082921)

[ghstack-poisoned]

Chillee committed:
Update on "[NNC] moved lowerings out of the TensorExprKernel and into independent functions"




moved lowerings out of the TensorExprKernel and into independent functions

Differential Revision: [D28082921](https://our.internmc.facebook.com/intern/diff/D28082921)

[ghstack-poisoned]

Chillee committed:
Update on "[NNC] moved lowerings out of the TensorExprKernel and into independent functions"




moved lowerings out of the TensorExprKernel and into independent functions

Differential Revision: [D28082921](https://our.internmc.facebook.com/intern/diff/D28082921)

[ghstack-poisoned]

Chillee committed:
Update on "[NNC] moved lowerings out of the TensorExprKernel and into independent functions"




moved lowerings out of the TensorExprKernel and into independent functions

Differential Revision: [D28082921](https://our.internmc.facebook.com/intern/diff/D28082921)

[ghstack-poisoned]
